### PR TITLE
[Merged by Bors] - chore(Algebra/Order/Group): change simp lemmas

### PR DESCRIPTION
**Pull request by astrainfinita**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by negiizhao** [comment]
!bench

File: Mathlib/Algebra/Order/Group/Defs.lean [hunk 1]
```diff
@@ -753,6 +753,8 @@ theorem div_le_iff_le_mul : a / c ≤ b ↔ a ≤ b * c := by
 #align div_le_iff_le_mul div_le_iff_le_mul
 #align sub_le_iff_le_add sub_le_iff_le_add
 
+attribute [simp] div_le_iff_le_mul
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Group/Defs.lean:756]: Please add a comment explaining why only the `div` version is simp

**Comment by negiizhao** [comment]
!bench

File: Mathlib/Data/Fin/Basic.lean [hunk 2]
```diff
@@ -1296,7 +1296,7 @@ theorem le_sub_one_iff {n : ℕ} {k : Fin (n + 1)} : k ≤ k - 1 ↔ k = 0 := by
   simp
 #align fin.le_sub_one_iff Fin.le_sub_one_iff
 
-@[simp]
+@[simp, nolint simpNF] -- lint timeout
```
Comment (j-loreaux) [inline Mathlib/Data/Fin/Basic.lean:1299]: This is a bit concerning. Any idea why it's occurring?

File: Mathlib/Algebra/Order/Sub/Defs.lean [hunk 3]
```diff
@@ -59,13 +59,14 @@ class OrderedSub (α : Type*) [LE α] [Add α] [Sub α] : Prop where
 
 section Add
 
-variable [Preorder α] [Add α] [Sub α] [OrderedSub α] {a b c d : α}
-
 @[simp]
-theorem tsub_le_iff_right : a - b ≤ c ↔ a ≤ c + b :=
+theorem tsub_le_iff_right [LE α] [Add α] [Sub α] [OrderedSub α] {a b c : α} :
```
[the remark this answers is not in the document]
Comment (negiizhao) [reply Mathlib/Algebra/Order/Sub/Defs.lean:63]: I don't know but some lemmas in `Algebra.Order.Group.Defs` use `LE`. After using `LE` here `simpNF` linter will ask to delete some `@[simp]` there.

File: Mathlib/Data/Fin/Basic.lean (hunk 2 above)
Comment (negiizhao) [reply Mathlib/Data/Fin/Basic.lean]: It's not needed now : )

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**